cmd/admin-heal-ops.go

```diff
@@ -608,25 +596,27 @@ func (h *healSequence) healSequenceStart() {
 
 	select {
 	case err, ok := <-h.traverseAndHealDoneCh:
+		if !ok {
+			return
+		}
+		h.mutex.Lock()
 		h.endTime = UTCNow()
-		h.currentStatus.updateLock.Lock()
-		defer h.currentStatus.updateLock.Unlock()
 		// Heal traversal is complete.
-		if ok {
+		if err == nil {
```

Comment:
I am not able to find the code which sends a nil error to h.traverseAndHealDoneCh, was it just buggy before ?

Reply:
so there has been quite a lot of bugs @vadmeste will update this PR once again traverseAndHealDoneCh will receive a channel close instead of nil in the current design. I am changing that again in a bit.

Reply:
Fixed